crates/autopilot/src/domain/settlement/mod.rs

```diff
@@ -185,3 +192,119 @@ pub enum Error {
+        0000000000000000000000000000000000000000000000000000000000000020
+        00000000000000000000000040a50cf069e992aa4536211b23f286ef88752187
+        0000000000000000000000000000000000000000000000000000000000000000
+        0000000000000000000000000000000000000000000000000000000000000060
+        0000000000000000000000000000000000000000000000000000000000000004
+        4c84c1c800000000000000000000000000000000000000000000000000000000
+        0000000000000000000000000000000000000000000000000000000000000003
+        0000000000000000000000000000000000000000000000000000000000000060
+        0000000000000000000000000000000000000000000000000000000000000140
+        0000000000000000000000000000000000000000000000000000000000000220
+        00000000000000000000000000000000be48a3000b818e9615d85aacfed4ca97
+        0000000000000000000000000000000000000000000000000000000000000000
+        0000000000000000000000000000000000000000000000000000000000000060
+        000000000000000000000000000000000000000000000000000000000000004f
+        0000000101010000000000000000063a508037887d5d5aca4b69771e56f3c92c
+        20840dd09188a65771d8000000000000002c400000000000000001c02aaa39b2
+        23fe8d0a0e5c4f27ead9083c756cc20000000000000000000000000000000000
+        000000000000000000000000c02aaa39b223fe8d0a0e5c4f27ead9083c756cc2
+        0000000000000000000000000000000000000000000000000000000000000000
+        0000000000000000000000000000000000000000000000000000000000000060
+        0000000000000000000000000000000000000000000000000000000000000044
+        a9059cbb000000000000000000000000c88deb1ce0bc4a4306b7f20be2abd28a
+        d3a5c8d10000000000000000000000000000000000000000000000001c5efcf2
+        c41873fd00000000000000000000000000000000000000000000000000000000
+        000000000000000000000000c88deb1ce0bc4a4306b7f20be2abd28ad3a5c8d1
+        0000000000000000000000000000000000000000000000000000000000000000
+        0000000000000000000000000000000000000000000000000000000000000060
+        00000000000000000000000000000000000000000000000000000000000000a4
+        022c0d9f00000000000000000000000000000000000000000000000000000000
+        000000000000000000000000000000000000000000000000000000ca2b0dae6c
+        b90dbc4b0000000000000000000000009008d19f58aabd9ed0d60971565aa851
+        0560ab4100000000000000000000000000000000000000000000000000000000
+        0000008000000000000000000000000000000000000000000000000000000000
+        0000000000000000000000000000000000000000000000000000000000000000
+        0000000000000000000000000000000000000000000000000000000000000000
+        000000000084120c"
+        )
+        .to_vec();
+
+        let domain_separator = eth::DomainSeparator(hex!(
+            "c078f884a2676e1345748b1feace7b0abee5d00ecadb6e574dcdd109a63e8943"
+        ));
+        let settlement = super::Settlement::new(&calldata.into(), &domain_separator).unwrap();
+        assert_eq!(settlement.trades.len(), 1);
```

Comment:
Since it seems fairly easy to add more tests with this setup do you think it makes sense to test a more complex solution to hopefully cover more corner cases?
Maybe Quasimodo has a nice solution with some CoWs we can take a look at. Not sure if this is super necessary, though.

Reply:
I think this depends on what we want to test. For surplus, we should probably test `Trade` with different values to cover different cases (buy/sell, partial) similar to what we did in `driver`.
For `Settlement` constructor, there might be space to add more tests, mostly related to error handling, but I also expect this part of code to change veeery rarely (maybe with new settlement contract?).